mesonbuild/environment.py

```diff
@@ -395,6 +395,27 @@ def detect_cpu(compilers: CompilersDict) -> str:
     # detect_cpu_family() above.
     return trial
 
+kernel_mappings = {'freebsd': 'freebsd',
```

Comment:
> I'm also wondering what happens when you run on debian/kfreebsd or gentoo/kfreebsd. @nirbheek do you happen to have a system you could test on? Otherwise I can spin up a VM

I don't, and I'm also away from my workstation till 21st, so it would be great if someone else could test it!

Reply:
AFAIK GNU/kfreebsd didn't really never went anywhere, I don't know how significant it is to add support for it. Debian is discontinuing the port https://lists.debian.org/debian-devel/2023/05/msg00306.html